### PR TITLE
docs - add content for new pxf read file as row feature

### PR DESCRIPTION
**Pull request by lisakowen**
pxf now supports reading one or more multi-line files where each file is a single row in the same external table.  the current pxf "using" docs are partitioned by external location: hadoop/objstore/jdbc.  within hadoop/objstore docs, we break the docs down by data format: text/avro/json/parquet...  this feature uses a previously-defined text profile (`*:text:multi`) to support text and JSON files.  adding content to the text page and JSON page doesn't make sense with this organization, so created a new page/topic for hadoop and a new page/topic for object store content.  as with the docs for other data formats, the HDFS page includes more detailed information, and the obj store page notes differences.

- the title "Reading a Multi-Line JSON or Text File into a Single Table Row" is kind of long, suggestions please.
- do we need to address quoting or escaping?

doc review site links:
- new hdfs topic:   http://docs-lisa-pxf2.cfapps.io/6-0/pxf/hdfs_fileasrow.html
- new objstore topic:  http://docs-lisa-pxf2.cfapps.io/6-0/pxf/objstore_fileasrow.html

### CODE REVIEW

File: gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+
+## <a id="prereq"></a>Prerequisites
+
+Ensure that you have met the PXF Hadoop [Prerequisites](access_hdfs.html#hadoop_prereq) before you attempt to read files from HDFS.
+
+
+## <a id="fileasrow"></a>Reading Multi-Line Text and JSON Files
+
+You can read single- and multi-line files into a single table row, including files with embedded linefeeds. If you are reading multiple JSON files, each file must be a complete record, and each file should contain the same record type.
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb]: should -> must

(?)

File: gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    ``` shell
+    $ echo 'Prague,Jan,101,4875.33
+Rome,Mar,87,1557.39
+Bangalore,May,317,8936.99
+Beijing,Jul,411,11600.67' > /tmp/file2.txt
+    ```
+
+    This file has multiple lines.
+
+4. Create a third text file. Open a file named `/tmp/file3.txt` in a text editor and copy/paste the following data into the file:
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb]: Why not use the `echo` as in the previous two steps?

File: gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+    ``` shell
+    $ hdfs dfs -put /tmp/file1.txt /data/pxf_examples/tdir
+    $ hdfs dfs -put /tmp/file2.txt /data/pxf_examples/tdir
+    $ hdfs dfs -put /tmp/file3.txt /data/pxf_examples/tdir
+    ```
+
+7. Log in to a Greenplum Database system and start the `psql` subsystem.
+
+8. Use the `hdfs:text:multi` profile to create an external table that references the `fileasrow` HDFS directory. For example:
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb]: Saying `fileasrow` here is a little weird, as that's not the actual name of the directory that was created.  Maybe just say "references the HDFS directory"?

File: gpdb-doc/markdown/pxf/objstore_fileasrow.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+| Google Cloud Storage    | gs |
+| Minio    | s3 |
+| S3    | s3 |
+
+The following syntax creates a Greenplum Database readable external table that references one or more text files in an object store:
+
+``` sql
+CREATE EXTERNAL TABLE <table_name>
+    ( <column_name> text|json | LIKE <other_table> )
+  LOCATION ('pxf://<path-to-hdfs-files>?PROFILE=hdfs:text:multi[&SERVER=<server_name>]&FILE_AS_ROW=true')
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/objstore_fileasrow.html.md.erb]: It doesn't seem like path-to-hdfs-files is correct here, is it?  Should be specific to the object store.

File: gpdb-doc/markdown/pxf/objstore_fileasrow.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  FORMAT 'CSV');
+```
+
+The specific keywords and values used in the [CREATE EXTERNAL TABLE](../ref_guide/sql_commands/CREATE_EXTERNAL_TABLE.html) command are described in the table below.
+
+| Keyword  | Value |
+|-------|-------------------------------------|
+| \<path&#8209;to&#8209;&#8209;files\>    | The absolute path to the directory or files in the object store. |
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/objstore_fileasrow.html.md.erb]: Keyword is mangled here.  Maybe should be just path-to-files

File: gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+The keywords and values used in this [CREATE EXTERNAL TABLE](../ref_guide/sql_commands/CREATE_EXTERNAL_TABLE.html) command are described in the table below.
+
+| Keyword  | Value |
+|-------|-------------------------------------|
+| \<path&#8209;to&#8209;files\>    | The absolute path to the directory or files in the HDFS data store. |
+| PROFILE    | The `PROFILE` keyword must specify `hdfs:text:multi`. |
+| SERVER=\<server_name\>    | The named server configuration that PXF uses to access the data. Optional; PXF uses the `default` server if not specified. |
+| FILE\_AS\_ROW=true    | The required option that instructs PXF to read each file into a single table row. |
+| FORMAT | The `FORMAT` must specify `'CSV'`.  |
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb]: not sure about your comment.  are you asking to remove the single quotes?

File: gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb (hunk 2 above)
Comment (lisakowen) [reply gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb]: i had thought there might be escaping issues, but that does not appear to be the case.  i updated this step to use echo.

File: gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,181 @@
+---
+title: Reading a Multi-Line Text or JSON File into a Single Table Row
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb]: I haven't tested this, but can you also use this feature to read XML data? 
https://gpdb.docs.pivotal.io/5180/admin_guide/query/topics/xml-data.html

Comment (frankgh) [reply gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb]: We can add it to the documentation later.

Comment (lisakowen) [reply gpdb-doc/markdown/pxf/hdfs_fileasrow.html.md.erb]: yes, lets add it later.

**Comment by lisakowen** [comment]
thanks for reviewing, @frankgh

**Pull request merged**